app/lib/communicator/communicator.js

```diff
@@ -223,6 +225,28 @@ function getSubmissionUrl( server ) {
     return ( server.lastIndexOf( '/' ) === server.length - 1 ) ? `${server}submission` : `${server}/submission`;
 }
 
+/**
+ * @param {Record<string, string | string[]>} [headers]
+ * @param {import('express').Request} [currentRequest]
+ */
+const getUpdatedRequestHeaders = ( headers = {}, currentRequest = getCurrentRequest() ) => {
+    const clientUserAgent = currentRequest?.headers['user-agent'];
+    const serverUserAgent = `Enketo/${config.version}`;
+    const userAgent = clientUserAgent == null
+        ? serverUserAgent
+        : `${serverUserAgent} ${clientUserAgent}`;
+
+    return {
+        ...headers,
+
+        // The Date header is forbidden to set programmatically client-side
```

Comment:
This comment has changed. So the issue is not setting the `Date` header, it's just setting it client-side?

Reply:
(Sounds right and the diff makes that seem pretty clear but want to totally confirm since the previous comment implied it wasn't allowed to be set anywhere)

Reply:
Yes, this was a clarification based on my understanding of the original and tracing its history. My hope was this clarification will help to short circuit future archaeology.